Documentation: INSTALL.md is outdated

Might be as simple as replacing all occurrences of `makeall.sh` with `make && make install && make image` - not sure.

Since ugprade to cmake I'm also seeing:

```
[  0%] Building CXX object Meta/Lagom/CMakeFiles/LagomCore.dir/__/__/Libraries/LibCore/TCPServer.cpp.o
/serenity/Libraries/LibCore/TCPServer.cpp:40:42: error: use of undeclared
      identifier 'SOCK_NONBLOCK'
    m_fd = socket(AF_INET, SOCK_STREAM | SOCK_NONBLOCK | SOCK_CLOEXEC, 0);
                                         ^
/serenity/Libraries/LibCore/TCPServer.cpp:40:58: error: use of undeclared
      identifier 'SOCK_CLOEXEC'
    m_fd = socket(AF_INET, SOCK_STREAM | SOCK_NONBLOCK | SOCK_CLOEXEC, 0);
                                                         ^
2 errors generated.
make[2]: *** [Meta/Lagom/CMakeFiles/LagomCore.dir/__/__/Libraries/LibCore/TCPServer.cpp.o] Error 1
make[1]: *** [Meta/Lagom/CMakeFiles/LagomCore.dir/all] Error 2
make: *** [all] Error 2
```

It looks like these are already defined. I've tried a clean repository and the same error persists.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,12 @@ add_custom_target(run
     USES_TERMINAL
 )
 
+add_custom_target(grub-image
+    COMMAND ${CMAKE_COMMAND} -E env "SERENITY_ROOT=${CMAKE_SOURCE_DIR}" ${CMAKE_SOURCE_DIR}/Meta/build-image-grub.sh
+    BYPRODUCTS ${CMAKE_BINARY_DIR}/_disk_image
+    USES_TERMINAL
+)
+
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -std=c++2a -fdiagnostics-color=always")
 
 include_directories(Libraries)

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -17,7 +17,7 @@ At present there is no real GPU support so don't expect OpenGL, Vulkan nor accel
 
 ## Creating a Serenity GRUB disk image
 
-Before creating a Serenity disk image, you need to build the OS as described in the [SerenityOS build instructions](https://github.com/SerenityOS/serenity/blob/master/Documentation/BuildInstructions.md). Follow those instructions up to and including running **./makeall.sh**. After the OS has built, run **sudo ./build-image-grub.sh** to replace the qemu **_disk_image** file in the Kernel directory with one that has GRUB2 installed that can be booted on a real PC. Alternately, you can run **sudo ./build-gpt-image-grub.sh** to create an image that uses GPT partitioning. 
+Before creating a Serenity disk image, you need to build the OS as described in the [SerenityOS build instructions](https://github.com/SerenityOS/serenity/blob/master/Documentation/BuildInstructions.md). Follow those instructions up to and including running **make install**. After the OS has built, run **sudo make grub-image** to replace the qemu **_disk_image** file in the Build directory with one that has GRUB2 installed that can be booted on a real PC.
 
 The final step is copying **_disk_image** onto the disk you wish to boot Serenity off using a command such as:
 
@@ -47,4 +47,4 @@ $ cu -s 57600 -l /dev/ttyUSB0
 
 ## Troubleshooting boot issues without a serial port
 
-Serenity is a graphical OS and thus boots directly to the desktop. If your computer doesn't have a serial port and it fails to boot, you can force Serenity to boot into text mode by editing **Kernel/Arch/i386/Boot/boot.S** and removing **| MULTIBOOT_VIDEO_MODE** from the end of the **multiboot_flags** line before (re)running **makeall.sh**.
+Serenity is a graphical OS and thus boots directly to the desktop. If your computer doesn't have a serial port and it fails to boot, you can force Serenity to boot into text mode by editing **Kernel/Arch/i386/Boot/boot.S** and removing **| MULTIBOOT_VIDEO_MODE** from the end of the **multiboot_flags** line before (re)running **make** and **make install**.

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -47,4 +47,4 @@ $ cu -s 57600 -l /dev/ttyUSB0
 
 ## Troubleshooting boot issues without a serial port
 
-Serenity is a graphical OS and thus boots directly to the desktop. If your computer doesn't have a serial port and it fails to boot, you can force Serenity to boot into text mode by editing **Kernel/Arch/i386/Boot/boot.S** and removing **| MULTIBOOT_VIDEO_MODE** from the end of the **multiboot_flags** line before (re)running **make** and **make install**.
+Serenity is a graphical OS and thus boots directly to the desktop. If your computer doesn't have a serial port and it fails to boot, you can force Serenity to boot into text mode by selecting the **SerenityOS (text mode)** option in GRUB.

--- a/Kernel/Arch/i386/Boot/boot.S
+++ b/Kernel/Arch/i386/Boot/boot.S
@@ -1,8 +1,7 @@
 .set MULTIBOOT_MAGIC,         0x1badb002
 .set MULTIBOOT_PAGE_ALIGN,    0x1
 .set MULTIBOOT_MEMORY_INFO,   0x2
-.set MULTIBOOT_VIDEO_MODE,    0x4
-.set multiboot_flags,         MULTIBOOT_PAGE_ALIGN | MULTIBOOT_MEMORY_INFO | MULTIBOOT_VIDEO_MODE
+.set multiboot_flags,         MULTIBOOT_PAGE_ALIGN | MULTIBOOT_MEMORY_INFO
 .set multiboot_checksum,      -(MULTIBOOT_MAGIC + multiboot_flags)
 
 .section .multiboot
@@ -19,12 +18,6 @@
 .long 0x00000000    /* load_end_addr */
 .long 0x00000000    /* bss_end_addr */
 .long 0x00000000    /* entry_addr */
-
-/* for MULTIBOOT_VIDEO_MODE */
-.long 0x00000000    /* mode_type */
-.long 1280          /* width */
-.long 1024          /* height */
-.long 32            /* depth */
 
 .section .stack, "aw", @nobits
 stack_bottom:

--- a/Meta/build-image-grub.sh
+++ b/Meta/build-image-grub.sh
@@ -90,8 +90,8 @@ echo "installing grub using $grub..."
 $grub --boot-directory=mnt/boot --target=i386-pc --modules="ext2 part_msdos" "${dev}"
 
 if [ -d mnt/boot/grub2 ]; then
-    cp grub-"${partition_scheme}".cfg mnt/boot/grub2/grub.cfg
+    cp "$SERENITY_ROOT"/Meta/grub-"${partition_scheme}".cfg mnt/boot/grub2/grub.cfg
 else
-    cp grub-"${partition_scheme}".cfg mnt/boot/grub/grub.cfg
+    cp "$SERENITY_ROOT"/Meta/grub-"${partition_scheme}".cfg mnt/boot/grub/grub.cfg
 fi
 echo "done"

--- a/Meta/grub-ebr.cfg
+++ b/Meta/grub-ebr.cfg
@@ -2,16 +2,16 @@ timeout=1
 
 menuentry 'SerenityOS (normal)' {
   root=hd0,5
-  multiboot /boot/kernel root=/dev/hda5
+  multiboot /boot/Kernel root=/dev/hda5
 }
 
 menuentry 'SerenityOS (No ACPI)' {
   root=hd0,5
-  multiboot /boot/kernel root=/dev/hda5 acpi=off
+  multiboot /boot/Kernel root=/dev/hda5 acpi=off
 }
 
 menuentry 'SerenityOS (with serial debug)' {
 	root=hd0,5
-  multiboot /boot/kernel serial_debug root=/dev/hda5
+  multiboot /boot/Kernel serial_debug root=/dev/hda5
 }
 

--- a/Meta/grub-ebr.cfg
+++ b/Meta/grub-ebr.cfg
@@ -5,6 +5,11 @@ menuentry 'SerenityOS (normal)' {
   multiboot /boot/Kernel root=/dev/hda5
 }
 
+menuentry 'SerenityOS (text mode)' {
+  root=hd0,5
+  multiboot /boot/Kernel boot_mode=text root=/dev/hda5
+}
+
 menuentry 'SerenityOS (No ACPI)' {
   root=hd0,5
   multiboot /boot/Kernel root=/dev/hda5 acpi=off

--- a/Meta/grub-gpt.cfg
+++ b/Meta/grub-gpt.cfg
@@ -5,6 +5,11 @@ menuentry 'SerenityOS (normal)' {
   multiboot /boot/Kernel root=/dev/hda2
 }
 
+menuentry 'SerenityOS (text mode)' {
+  root=hd0,2
+  multiboot /boot/Kernel boot_mode=text root=/dev/hda2
+}
+
 menuentry 'SerenityOS (No ACPI)' {
   root=hd0,2
   multiboot /boot/Kernel root=/dev/hda2 acpi=off

--- a/Meta/grub-gpt.cfg
+++ b/Meta/grub-gpt.cfg
@@ -2,15 +2,15 @@ timeout=1
 
 menuentry 'SerenityOS (normal)' {
   root=hd0,2
-  multiboot /boot/kernel root=/dev/hda2
+  multiboot /boot/Kernel root=/dev/hda2
 }
 
 menuentry 'SerenityOS (No ACPI)' {
   root=hd0,2
-  multiboot /boot/kernel root=/dev/hda2 acpi=off
+  multiboot /boot/Kernel root=/dev/hda2 acpi=off
 }
 
 menuentry 'SerenityOS (with serial debug)' {
 	root=hd0,2
-  multiboot /boot/kernel serial_debug root=/dev/hda2
+  multiboot /boot/Kernel serial_debug root=/dev/hda2
 }

--- a/Meta/grub-mbr.cfg
+++ b/Meta/grub-mbr.cfg
@@ -5,12 +5,17 @@ menuentry 'SerenityOS (normal)' {
   multiboot /boot/Kernel root=/dev/hda1
 }
 
+menuentry 'SerenityOS (text mode)' {
+  root=hd0,1
+  multiboot /boot/Kernel boot_mode=text root=/dev/hda1
+}
+
 menuentry 'SerenityOS (No ACPI)' {
   root=hd0,1
   multiboot /boot/Kernel root=/dev/hda1 acpi=off
 }
 
 menuentry 'SerenityOS (with serial debug)' {
-	root=hd0,1
+  root=hd0,1
   multiboot /boot/Kernel serial_debug root=/dev/hda1
 }

--- a/Meta/grub-mbr.cfg
+++ b/Meta/grub-mbr.cfg
@@ -2,15 +2,15 @@ timeout=1
 
 menuentry 'SerenityOS (normal)' {
   root=hd0,1
-  multiboot /boot/kernel root=/dev/hda1
+  multiboot /boot/Kernel root=/dev/hda1
 }
 
 menuentry 'SerenityOS (No ACPI)' {
   root=hd0,1
-  multiboot /boot/kernel root=/dev/hda1 acpi=off
+  multiboot /boot/Kernel root=/dev/hda1 acpi=off
 }
 
 menuentry 'SerenityOS (with serial debug)' {
 	root=hd0,1
-  multiboot /boot/kernel serial_debug root=/dev/hda1
+  multiboot /boot/Kernel serial_debug root=/dev/hda1
 }

